### ISSUE TITLE
Update dataflow to constrain memory based on usage. Also function table detection.

### ISSUE THREAD

**Issue by krx** [Vector35/binaryninja-api#967]
This is very similar to #966, but instead a function pointer is loaded from a static table. Ideally each function in this table could be defined automatically

Example binary: https://github.com/trailofbits/mcsema/blob/master/tests/test_suite/amd64/linux/bin/ls_amd64

Example of a function pointer being loaded from a table: `sub_404f90 : 0x405094`
Broken jump table `sub_407790 : 0x4077ad`

**Comment by bpotchik** [comment]
Solving for rax at 0x405094 is not within the capabilities of our system.

For 0x4077ad, I added the ability to solve for undetermined values of memory versions based on usage. This allows for branch sensitive constraints to be generated for these types of jump tables. We may end up converting this to a heuristic based analysis option in the future. If you notice, there is a potential toctou bug in this binary.

Fixed in build 1.1.1074-dev.
